asciidoc/courses/gds-getting-started-with-graph-data-science/modules/1-gds-foundations/lessons/2-understanding-graph-data-science/lesson.adoc

```diff
@@ -0,0 +1,185 @@
+
+This query returns the same results as a graph visualization.
+
+[source,cypher,role=nocopy]
+----
+MATCH path = SHORTEST 10 (chaz:Person {name: 'Charlie Chaplin'})
+    -[:ACTED_IN]-+ 
+        (kevin:Person {name:'Kevin Bacon'})
+RETURN path
+----
+
+Both queries finish in well under a second. Even for something this simple, a tabular approach would require serious refactoring or a series of joins to get the same data.
+
+When you analyze the structure of information—not just properties alone—you can gain insights about user behaviors, identify supply chain bottlenecks, predict service failures, and more.
+
+
+== How graph data science differs from data science
+
+In the previous example, you searched for paths between actors using Cypher. GDS takes this many steps further. 
+
+Imagine you want to create a recommendation engine for a content platform, but all your transactional data is tabular.
+
+[opts="header",cols="1,1,1"]
+|===
+| user_id      | item_id         | user_rating
+| user_01      | item_422437     | 4
+| user_01      | item_752639     | 5
+| user_01      | item_782392     | 3
+| user_01      | item_125565     | 2
+|===
+
+Just because a user enjoyed three Liam Neeson action films doesn't mean they love Liam Neeson -- or even action movies in general. Recommending movies based only on metadata similarities and user ratings is both naive and extremely common. 
+
+An effective recommendation engine would analyze user behaviors alongside user and movie metadata. It would also account for how those data interact -- but let's not get ahead of ourselves.
+
+To assess behavioral similarities between users in a tabular format, you'd need to reshape the dataset into multiple tables, each with a specific purpose.
+
+[opts="header",cols="1,1,1,1,1,1"]
+|===
+| user_id | item_422437 | item_752639 | item_782392 | item_125565 | ...
+| user_01 | 4           | 5           | 3           | 2           |
+| user_02 | 3           | 5           | 1           | 1           |
+| user_03 | 4           | 4           | 4           | 3           |
+| user_04 | 1           | 1           | 2           | 1           |
+|===
+
+Then, to account for various patterns, you would need to use expensive JOIN operations, which can only get you so far.
+
+Graph data represents relationships explicitly. You don't need one table to connect users to actors, another to connect actors to movies, and another to relate metadata to user journeys. You need only one dataset to store and retrieve all of these perspectives at once.
+
+The image below shows Liam Neeson's node in the movie graph—he's the large red one. The mid-sized orange nodes are movies he appeared in. The smaller blue nodes are users who rated those movies. 
+
+image::images/lesson_2_liam_neeson.png[a graph network showing the connections between Liam Neeson, to movies he appeared in and Users who rated those movies]
+
+Run this query to explore the graph yourself.
+
+[source,cypher,role=nocopy]
```

Comment:
Titles on code blocks and admonitions in case someone is scanning

```
[source,cypher,role=nocopy]
.Users who rated movies starring Liam Neeson
----
MATCH path = (a:Actor {name: 'Liam Neeson'})
    -[:ACTED_IN]
        ->(m:Movie)<-
    [:RATED]-(u:User)
RETURN path
----
```